rest-server/src/templates/yarnContainerScript.mustache

```diff
@@ -73,6 +73,92 @@ docker login --username ${cred[1]} --password ${cred[2]} ${cred[0]} \
   || { echo 'Authorized failed' ; exit 1; }
 {{/ jobData.authFile }}
 
+
+# Get container host list from framework launcher webservice
+function get_host_list()
+{
+  delay_time=1
+  while true; do
+    taskrole_statuses=$(curl -i -L -H "Accept: application/json" {{{ frameworkInfoWebhdfsUri }}} | jq -r ".aggregatedFrameworkStatus.aggregatedTaskRoleStatuses")
+    jq -re ".[].taskStatuses.taskStatusArray[].containerIp" <<< $(echo $taskrole_statuses) > /dev/null
+    if [ $? -eq 0 ]; then
+      break
+    fi
```

Comment:
Also print log for each failure, at least include which URI is accessed.

Reply:
enabled `set -x` at the beginning of this script to record execution details which will be redirected to hdfs

Reply:
why need this?
Can we just printf ?

---
In reply to: [192627750](https://github.com/Microsoft/pai/pull/467#discussion_r192627750) [](ancestors = 192627750)